{
    "source": "BDSA",
    "name": "BDSA-2020-1246",
    "title": "SQLite Vulnerable to Denial-of-Service (DoS) via Use After Free in 'fts3EvalNextRow' Function",
    "description": "SQLite is vulnerable to denial-of-service (DoS) due to a use after free issue in the FTS3 virtual table module. An attacker could exploit this vulnerability by supplying a system with maliciously crafted input.",
    "technicalDescription": "The issue exists in the function `fts3EvalNextRow()` within the file `ext/fts3/fts3.c`. Improper management of the end-of-file pointers `pRight->bEof` and `pLeft->bEof` could lead to a use after free condition.",
    "publishedDate": "2020-05-28T13:44:32.391Z",
    "updatedDate": "2020-10-26T14:27:10.579Z",
    "disclosureDate": "2020-05-16T00:00:00.000Z",
    "solution": "Fixed in **[3.32.0](https://sqlite.org/download.html)** by [this](https://sqlite.org/src/info/0d69f76f0865f962) commit.",
    "severity": "MEDIUM",
    "cvss2": {
        "baseScore": 5.0,
        "impactSubscore": 2.9,
        "exploitabilitySubscore": 10.0,
        "severity": "LOW",
        "accessVector": "NETWORK",
        "accessComplexity": "LOW",
        "authentication": "NONE",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "PARTIAL",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 3.7
        },
        "vector": "(AV:N/AC:L/Au:N/C:N/I:N/A:P/E:U/RL:OF/RC:C)"
    },
    "cvss3": {
        "baseScore": 7.5,
        "impactSubscore": 3.6,
        "exploitabilitySubscore": 3.9,
        "severity": "MEDIUM",
        "attackVector": "NETWORK",
        "attackComplexity": "LOW",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "HIGH",
        "privilegesRequired": "NONE",
        "scope": "UNCHANGED",
        "userInteraction": "NONE",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 6.5
        },
        "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H/E:U/RL:O/RC:C"
    },
    "useCvss3": true,
    "classifications": [],
    "zeroDay": false,
    "underReview": false,
    "parentAdvisory": false,
    "workaround": "",
    "vendorFixDate": "2020-05-16T00:00:00.000Z",
    "bdsaTags": [],
    "overallScore": 6.5,
    "_meta": {
        "allow": [],
        "href": "https://trendmicro.app.blackduck.com/api/vulnerabilities/BDSA-2020-1246",
        "links": [
            {
                "rel": "cwes",
                "href": "https://trendmicro.app.blackduck.com/api/cwes/CWE-416"
            },
            {
                "rel": "related-vulnerability",
                "href": "https://trendmicro.app.blackduck.com/api/vulnerabilities/CVE-2020-13630",
                "label": "NVD"
            },
            {
                "rel": "bdsa-ranges",
                "href": "https://trendmicro.app.blackduck.com/api/vulnerabilities/BDSA-2020-1246/ranges"
            },
            {
                "rel": "reference",
                "href": "https://bugs.chromium.org/p/chromium/issues/detail?id=1080459",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://sqlite.org/download.html",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "reference",
                "href": "https://sqlite.org/src/info/0d69f76f0865f962",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://sqlite.org/src/info/0d69f76f0865f962",
                "label": "PATCH"
            }
        ]
    }
}